src/msgpack/unpacker.d

```diff
@@ -196,7 +196,7 @@ struct Unpacker
         if (0x00 <= header && header <= 0x7f) {
             value = cast(T)header;
         } else if (0xe0 <= header && header <= 0xff) {
-            value = -(cast(T)-header);
+            value = cast(T)cast(byte)header;
```

Comment:
https://github.com/msgpack/msgpack-d/pull/99/files << take a look at this, I tried that one myself first but unittests were broken so I had to rethink it :-D

Since that PR is still not handled you can get that version of msgpack-d here: https://github.com/causal-rt/msgpack-d

Reply:
this looks good.

Reply:
@RalphBariz  I've took. I guess your solution is more general, my one handles the specific case only, but due to this fact is much simpler. 
But why x64 unittest fails?

Reply:
I also tried to remove the -(-()) double negation and faced broken unittests when running in 64bit. Because of that I reintroduced the -- and made that wild casting stuff.